app/services/invoices/calculate_fees_service.rb

```diff
@@ -197,9 +202,17 @@ def create_coupon_credit
 
         next if applied_coupon.coupon.fixed_amount? && applied_coupon.amount_currency != currency
 
-        credit_result = Credits::AppliedCouponService.new(
-          invoice:, applied_coupon:,
-        ).create
+        base_amount_cents = if applied_coupon.coupon.limited_plans?
+          coupon_related_fees = coupon_fees(applied_coupon)
+
+          next unless coupon_related_fees.exists?
```

Comment:
Not sure exactly how the process is supposed to behave if no fee are found for the applied coupon. Should we return 0 instead of nothing?

Reply:
closing this one as per discussion with Mike